packages/docusaurus/lib/client/exports/ComponentCreator.js

```diff
@@ -8,62 +8,67 @@
 import React from 'react';
 import Loadable from 'react-loadable';
 import Loading from '@theme/Loading';
-import routesAsyncModules from '@generated/routesAsyncModules';
+import routesChunkNames from '@generated/routesChunkNames';
 import registry from '@generated/registry';
 
 function ComponentCreator(path) {
-  const modules = routesAsyncModules[path];
+  /* chunkNames are mapping of the route module -> webpack chunk name
+  Example:
+    {
+      "component": "component---theme-pages-767",
+      "content": {
+        "1": "module---06-c-47f",
```

Comment:
Why is this changed to an object? It should remain as an array. Objects have no ordering but arrays do and over here the ordering matters.

Reply:
Actually there is no logic change here. I just renamed `routesAsyncModules` to `routesChunkName` and put the example structure here.

```js
/* chunkNames are mapping of the route module -> webpack chunk name
  Example:
    {
      "component": "component---theme-pages-767",
      "content": {
        "1": "module---06-c-47f",
      }
    }
  */
```

To describe that later on it will be converted to
```js
/* Prepare opts data that react-loadable needs
  https://github.com/jamiebuilds/react-loadable#declaring-which-modules-are-being-loaded
  Example: (with above example as input)
  - optsLoader: 
  {
    component: () => import('./Pages.js'),
    content.1: () => import('./doc1.md'),
  }
  - optsModules: ['./Pages.js', './doc1.md']
  - optsWebpack: [require.resolveWeak('./Pages.js'), require.resolveWeak('./doc1.md')]
  */
```

You are the one who made the structure like this 😃 https://github.com/facebook/Docusaurus/pull/1366/files

Reply:
I guess it's not wrong to put that as an example, but the key `1` is misleading. I thought you changed my array to keys 😓